11-type-constructors/algo/src/Internal.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Internal where
+
+import Common
+import Control.Exception.Base
+import qualified External as E
+
+-- The internal type language
+-- (following ``convention'', we speak of terms and types, not types and kinds
+-- in the context of normalization)
+-- TODO: In the future, should contain kind casts for a full gradual
```

Comment:
what can't currently be expressed?

Reply:
if you write something like
```
type T = Int in
type V = λt::KHole.t Bool in
type Bogus = V T in
...
```
kind casts would let you express why Bogus is bogus